uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Upgrader/Steps/1004.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+trait CRM_HRCore_Upgrader_Steps_1004 {
+
+  /**
+   * Upgrader to set 'Home' as default location and set all 'Home' addresses as
+   * primary.
+   *
+   * @return bool
+   */
+  public function upgrade_1004() {
+    $this->up1004_setHomeLocationAsDefault();
+    $this->up1004_setAllHomeAddressesAsPrimary();
+
+    return true;
+  }
+
+  private function up1004_setHomeLocationAsDefault() {
+    civicrm_api3('LocationType', 'get', [
+      'sequential' => 1,
+      'name' => 'Home',
+      'api.LocationType.create' => [
+        'id' => '$value.id',
+        'is_default' => 1,
+        'is_reserved' => 1,
+      ],
+    ]);
+  }
+
+  private function up1004_setAllHomeAddressesAsPrimary() {
+    $homeLocation = civicrm_api3('LocationType', 'getsingle', [
+      'name' => "Home",
+    ]);
+
+    $contactsWithNonPrimaryHomeAddress = civicrm_api3('Address', 'get', [
+      'sequential' => 1,
+      'return' => array('contact_id'),
+      'location_type_id' => $homeLocation['id'],
+      'is_primary' => 0,
+    ]);
+
+    $contactIDs = [];
+    foreach ($contactsWithNonPrimaryHomeAddress['values'] as $currentContact) {
+      $contactIDs[] = $currentContact['contact_id'];
+    }
+
+    if (!empty($contactIDs)) {
+      $query = "
+        UPDATE civicrm_address
+        SET is_primary = CASE WHEN location_type_id = {$homeLocation['id']} THEN 1 ELSE 0 END
+        WHERE contact_id IN (" . implode(',', $contactIDs) . ")
+      ";
```

Comment:
@MiyaNoctem the point about using the API is that we should it as much as possible (when dealing with CiviCRM core entities) and we should resort to raw SQL queries only when there's no other option. I'm aware that using the API is not a performant option, but this upgrader will be executed only once on each client so I think it's fine.

Now, about setting and unsetting the is_primary value. You don't need to do that manually. If you use the API to set an Address as primary, it will automatically unset it for all the other addresses of the contact. So I think you can do everything with just:

```php
$result = civicrm_api3('Address', 'get', array(
  'sequential' => 1,
  'location_type_id' => "Home",
  'is_primary' => 0,
  'api.Address.create' => array('id' => "\$value.id", 'is_primary' => 1),
));
```

(I didn't really test this, so it might not work)